plugins/woocommerce/client/legacy/js/frontend/woocommerce.js

```diff
@@ -194,9 +196,54 @@ function refresh_sorted_by_live_region () {
 	}
 }
 
+function init_wc_float_labels () {
+	var inputs_with_wc_float_label = document.querySelectorAll( '.input-with-wc-float-label' );
+
+	function get_label_from_input_id( input_id ) {
+		return document.querySelector( 'label[for="' + input_id + '"]' );
+	}
+
+	inputs_with_wc_float_label.forEach( function( input ) {
+		var label = get_label_from_input_id( input.id );
+
+		if ( label ) {
+			var fake_placeholder = label.parentElement.querySelector( '[class*="placeholder"]' );
```

Comment:
The `select2` library creates a fake placeholder like this:

```html
<span class="select2-selection__placeholder">Select an option…</span>
```
![Screenshot 2024-10-25 at 17 32 43](https://github.com/user-attachments/assets/0ef075c2-08d8-470f-881d-5f262ea3f34c)

In this case, the `label` element needs to get the `is-active` class so it doesn't overlap with the fake placeholder.